libs/mini-rx-store/src/lib/extensions/command/command-combine-reducers.ts

```diff
@@ -0,0 +1,65 @@
+import { Action, Reducer, ReducerDictionary } from '../../models';
+import {
+    addManyCmds,
+    cleanCmds,
+    Cmd,
+    hasCmds,
+    SliceReducerWithCmds,
+    withCmd,
+} from './command-models';
+
+export function commandCombineReducers<T extends object = {}>(
```

Comment:
As long as FeatureStore keeps working, it should be fine. 
If normal reducers still work, then the FeatureStore reducer will work as well. 

FeatureStore is less suitable for customization. Although it uses the extensions which are passed to configureStore. But the internal creation of a setState Action and the FeatureStore reducer are rather set in stone.

_Re. where/when to pass an alternative combineReducers function_: Indeed the constructor uses combineReducers and subscribes directly. Subscription could be delayed until the first reducer is added, but on the other hand the first run of the ` this.reducerStateSource.subscribe` can simply be ignored: Nothing happens with the calculated reducer until the first Action is emitted.
It is only important that the combineReducers fn is properly set before adding reducers and metaReducers and before the first Action is dispatched (MiniRxActionType.INIT_STORE).

Reply:
_Re. where/when to pass combineReducers fn._: so are we saying that `reducerStateSource` subscription could be done once and for all before [this line](https://github.com/spierala/mini-rx-store/blob/b141aea451e18e792ebb10c9a2cd1ab32ae0925a/libs/mini-rx-store/src/lib/store-core.ts#L83), with the `combineReducers` fn taken either from store configuration or from a default?
Otherwise, one should store first subscription at construction time, and then at configuration time could potentially unsubscribe and subscribe again. But it seems little more messy, and if there's no need to subscribe before configuration...

Reply:
You must keep in mind, configureStore is an optional thing.

FeatureStores should work also if configureStore is never called.
Therefore the reducerStateSource subscription happens independently of configureStore in the constructor.
A FeatureStore will also add a reducer and dispatch an initial Action to StoreCore.
For that reason, everything has to be wired up in the constructor, even if configureStore was never called.

Theoretically, reducerStateSource could better be an ReplaySubject(1) (without the initial/useless emission). But I chose for BehaviorSubject because it has getValue and it is more lightweight than ReplaySubject. 
reducerStateSource with skip(1) would also be an option. But again, the first run of the this.reducerStateSource.subscribe callback can be ignored and is ignored now. The calculated reducer is not relevant until the first Action is dispatched.
You just have to make sure that the commandCombineReducers is set before reducers, metaReducers are added and before the first Action is dispatched. And that is indeed before [this line](https://github.com/spierala/mini-rx-store/blob/b141aea451e18e792ebb10c9a2cd1ab32ae0925a/libs/mini-rx-store/src/lib/store-core.ts#L83).

I hope that makes sense :)

Reply:
Let's say we proceed with storing subscription in constructor, and allowing to re-subscribe in `config` if a custom combineReducers fn is passed in the options.

Re-subscribing to `this.reducerStateSource` doesn't seem enough though. The subscribe() body also alters `reducer` local variable, which in constructor is used to "Listen to the Actions Stream and update state accordingly". So basically also the `this.actions$` subscription should be re-done.

Is that right?

Reply:
I was able to reproduce the failing tests. But I managed to let the tests be green like that (I changed T to AppState):

![Screenshot 2022-11-23 at 17 16 26](https://user-images.githubusercontent.com/1272446/203596401-b7ef351b-89dd-4bd0-9300-ab1f5844b1bf.png)

Please do not worry about resubscribing stuff. 
I am quite sure that everything will work as expected if you overwrite the default combineReducers before adding the reducers and metaReducers and before dispatching MiniRxActionType.INIT_STORE in `configureStore`.

Adding reducers and metaReducers in configureStore will trigger a recalculation of `reducer` in the constructor. 
The recalculation will use then the overwritten combineReducers fn. 
Until the first Action is dispatched, `reducer` will be calculated using the overwritten combineReducers fn. 
Can you try it? If you encounter issues, then we will find a solution.

Reply:
By the time I read the comment, we already went for the "complicated" way with re-subscribe. Anyway, we turned back for the simpler solution mention above, i.e. to just overwrite a combineReducer fn private field.

In a separate branch, we tried adding the same vanilla demo app and this time command extension was provided in the app, as a 3rd-party extension, out of mini-rx-store library. And all goes fine.

PR is on the way, please have a look at it.

Reply:
Sorry, I had to insist on the simple solution :)
MiniRx tries to be simple and lightweight. 
Your PR looks great!

Reply:
no worries.